python/feathub/feature_tables/feature_table.py

```diff
@@ -73,6 +79,7 @@ def __init__(
         self.system_name = system_name
         self.properties = properties
         self.schema = schema
+        self._is_bounded = is_bounded
```

Comment:
Since `is_bounded` is a feature that can be derived from existing fields and the actual class type, would it be simpler to keep `FeatureTable` compact without adding this field? We keep the `is_bounded` only for those classes that actually need it (e.g. KafkaSource).

We would also skip passing `is_bounded=True` for many classes.